syntax = "proto3";
import "google/protobuf/empty.proto";

package audioservice;

service AudioService {
    rpc RunCmd (Command) returns (CmdOutput);
    rpc Play (File) returns (stream AudioData);
    rpc GetAudioMetadata (File) returns (AudioMetadata);
    rpc StreamAudio (google.protobuf.Empty) returns (stream AudioData);
}

message Command {
    // Command to be run
    string cmd = 1;
    // Optional arg
    string arg = 2;
}

message File {
    string name = 1;
}

message CmdOutput {
    string message = 1;
}

message AudioMetadata {
  uint32 AverageBytesPerSecond = 1;
  uint32 BitsPerSample = 2;
  uint32 BlockAlign = 3;
  uint32 Channels = 4;
  uint32 ExtraSize = 5;
  uint32 SampleRate = 6;
}

message AudioData {
    oneof response {
        AudioMetadata MetaData = 1;
        bytes RawData = 2;
    }
}
